bindings/rust/s2n-tls/src/fingerprint.rs

```diff
@@ -0,0 +1,497 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Methods to fingerprint ClientHellos.
+//!
+//! See [the C API documentation](https://github.com/aws/s2n-tls/blob/main/api/unstable/fingerprint.h).
+
+use crate::{
+    client_hello::ClientHello,
+    error::{Error, Fallible},
+    ffi::*,
+};
+
+use core::ptr::NonNull;
+
+#[non_exhaustive]
+#[derive(Copy, Clone)]
+pub enum FingerprintType {
+    JA3,
+}
+
+const MD5_HASH_SIZE: u32 = 16;
+
+impl From<FingerprintType> for s2n_tls_sys::s2n_fingerprint_type::Type {
+    fn from(value: FingerprintType) -> Self {
+        match value {
+            FingerprintType::JA3 => s2n_tls_sys::s2n_fingerprint_type::FINGERPRINT_JA3,
+        }
+    }
+}
+
+/// A fingerprint operation.
+pub struct Fingerprint<'a>(&'a mut NonNull<s2n_fingerprint>);
+
+impl Fingerprint<'_> {
+    /// Size of the fingerprint hash.
+    /// See s2n_fingerprint_get_hash_size in [the C API documentation](https://github.com/aws/s2n-tls/blob/main/api/unstable/fingerprint.h).
+    pub fn hash_size(&self) -> Result<u32, Error> {
+        let mut hash_size = 0;
+        unsafe { s2n_fingerprint_get_hash_size(self.0.as_ptr(), &mut hash_size).into_result() }?;
+        Ok(hash_size)
+    }
+
+    /// Calculate the fingerprint hash as a hex string.
```

Comment:
Doc ask: could we add an example fingerprint hash and fingerprint raw string in the respective methods? I don't know how useful it is, but personally I find that have example outputs makes things easier to reason about.